src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -6424,13 +6450,32 @@ public object VisitIndexExpression(IndexExpressionAst indexExpressionAst)
             // expression.
             if (arrayLiteral != null && arrayLiteral.Elements.Count > 1)
             {
-                return DynamicExpression.Dynamic(
-                                            PSGetIndexBinder.Get(arrayLiteral.Elements.Count, constraints),
-                                            typeof(object),
-                                            arrayLiteral.Elements.Select(CompileExpressionOperand).Prepend(targetExpr));
+                var dynamicExprArrayLiteral = DynamicExpression.Dynamic(
+                    PSGetIndexBinder.Get(arrayLiteral.Elements.Count, constraints),
+                    typeof(object),
+                    arrayLiteral.Elements.Select(CompileExpressionOperand).Prepend(targetExpr));
+
+                return indexExpressionAst.NullConditionalAccess ? GetNullCheckExpressionBlock(targetExpr, dynamicExprArrayLiteral) : dynamicExprArrayLiteral;
+
             }
 
-            return DynamicExpression.Dynamic(PSGetIndexBinder.Get(1, constraints), typeof(object), targetExpr, CompileExpressionOperand(index));
+            var dynamicExprFromBinder = DynamicExpression.Dynamic(PSGetIndexBinder.Get(1, constraints), typeof(object), targetExpr, CompileExpressionOperand(index));
+            return indexExpressionAst.NullConditionalAccess ? GetNullCheckExpressionBlock(targetExpr, dynamicExprFromBinder) : dynamicExprFromBinder;
+        }
+
+        internal static Expression GetNullCheckExpressionBlock(Expression targetExpr, Expression memberAccessExpression)
```

Comment:
How about rename it to `GetNullConditionalWrappedExpression`? Also it's better to make this method private.

Reply:
Fixed,